### PR TITLE
fix: A requesting top variably expressed genes, no longer limit to CG…

### PR DESCRIPTION
**Pull request by xzhou82**
…C set

## Description

Congyu please test with local ~~GFF~~ PP against new api currently in qa; you need to connect gdc vpn
please find out time taken, and assess gene sets returned for all cases, gliomas, HGG, if they makes sense

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by congyu-lu** [comment]
> …C set
> 
> ## Description
> Congyu please test with local GFF against new api currently in qa; you need to connect gdc vpn please find out time taken, and assess gene sets returned for all cases, gliomas, HGG, if they makes sense
> 
> ## Checklist
> [Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
> 
> * [ ]  Tests: added and/or passed unit and integration tests, or N/A
> * [ ]  Todos: commented or documented, or N/A
> * [ ]  Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

Tested with local GFF using qa-orange VPN
1. all cases: took about 70 seconds to Loading genes that are top variably expressed in current cohort...
(until showing "Processing data...")
image

2. After adding glioma filter:
image

**Comment by congyu-lu** [comment]
> > …C set
> > ## Description
> > Congyu please test with local GFF against new api currently in qa; you need to connect gdc vpn please find out time taken, and assess gene sets returned for all cases, gliomas, HGG, if they makes sense
> > ## Checklist
> > [Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
> > 
> > * [ ]   Tests: added and/or passed unit and integration tests, or N/A
> > * [ ]   Todos: commented or documented, or N/A
> > * [ ]   Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A
> 
> Tested with local GFF using qa-orange VPN
> 
> 1. all cases: took about 70 seconds to Loading genes that are top variably expressed in current cohort...
>    (until showing "Processing data...")
> 
> image
> 2. After adding glioma filter:
> 
> image

Need to update the legend label here, it still shows CGC genes only
image

**Comment by xzhou82** [comment]
70s is not okay.. how much time is it for gliomas?

**Comment by congyu-lu** [comment]
> 70s is not okay.. how much time is it for gliomas? and this doesn't seem right. Phil has indicated the new api should be much faster than this. please doublecheck if it is querying against the qa environment. also i don't think you have to test it with local GFF. just pp at http://localhost:3000/ should be good

Tested with qa-orange VPN, for both local GFF and http://localhost:3000: from "_Loading genes that are top variably expressed in current cohort_" to "_Processing data..._": All cases took about 70s, glioma took about 60s.

Also in here this note needs to be updated: _Genes are selected from 738 Cancer Gene Census genes._
image

**Comment by xzhou82** [comment]
thanks will remember to fix those once the api performance issue is resolved

File: server/routes/termdb.topVariablyExpressedGenes.ts [hunk 1]
```diff
@@ -174,41 +174,10 @@ function gdcValidateQuery(ds: any, genome: any) {
 	}
 
 	function getGeneSelectionArg(q: any, caseLst: any) {
-		//to hide messy logic during testing phase
-
-		/* when api performance issue is resolved, return this
```
Comment (xzhou82) [inline server/routes/termdb.topVariablyExpressedGenes.ts:179]: can add a temp console log just to verify it no longer restricts to cgs

**Comment by xzhou82** [comment]
close this PR since api is not fast and cannot be the default; still use cgc as default but will allow user to deselect it

**Pull request closed without merging**